cranelift-codegen/meta/src/default_map.rs

```diff
@@ -0,0 +1,28 @@
+use std::collections::HashMap;
+use std::hash::Hash;
+
+/// A standard map with an additional get_or_default method that allows to create default entries
+/// when a key is looked up for the first time.
+pub struct DefaultMap<K, V> {
```

Comment:
Note: This new type seems to be unused as part of part 1.

Reply:
And you're correct; I haven't bother to use a trailing underscore for unused entities, since it would add silly changes to the subsequent PRs. Happy to do that if that's preferable.

Reply:
I am fine with the current changes.